modules/http/tests/http_validation.cc

```diff
@@ -37,8 +47,107 @@ BOOST_AUTO_TEST_CASE(test_parse_url)
 	testGString.allocated_len=8;
 	testGString.len=3;
 	strcpy(testPlainString,"GET");
-	proxyFake.request_url=g_string_new("http://example.com");
-	proxyFake.request_method=&testGString;
-	http_split_request(&proxyFake,line,33);
+	proxyFake->request_method=&testGString;
+
+	http_split_request(proxyFake,line,33);
 	BOOST_CHECK(testPlainString[8]=='b');
 }
+
+typedef struct {
+	const char* input_line;
+	const char* expected_method;
+	const char* expected_url;
+	const char* expected_version;
+} TestCase;
+
+TestCase testCases[] = {
+		{ "GET http://example.com HTTP/1.0", "GET", "http://example.com", "HTTP/1.0" },
+		{ "GET  http://example.com HTTP/1.0", "GET", "http://example.com", "HTTP/1.0" },
+		{ "GET                http://example.com HTTP/1.0", "GET", "http://example.com", "HTTP/1.0" },
+		{ "GET                http://example.com  HTTP/1.0", "GET", "http://example.com", "HTTP/1.0" },
+		{ "GET                http://example.com                            HTTP/1.0", "GET", "http://example.com", "HTTP/1.0" },
+		{ "GET                http://example.com                            HTTP/1.0 ", "GET", "http://example.com", "HTTP/1.0" },
+		{NULL, NULL,NULL, NULL}
+};
+
+BOOST_AUTO_TEST_CASE(test_correct_line_is_parsed_correctly)
+{
+	int n = 0;
+	do {
+		const char* inputLine = testCases[n].input_line;
+		if (inputLine == NULL) {
+			break;
+		}
+		printf("%s\n",inputLine);
+		HttpProxy* proxyFake = new_proxy();
+		gboolean returnValue = http_split_request(proxyFake,inputLine,strlen(inputLine));
+
+		BOOST_CHECK(TRUE==returnValue);
+		BOOST_CHECK(TRUE==g_string_equal(proxyFake->request_method,g_string_new(testCases[n].expected_method)));
+		BOOST_CHECK(TRUE==g_string_equal(proxyFake->request_url,g_string_new(testCases[n].expected_url)));
+		BOOST_CHECK(0==strcmp(proxyFake->request_version,testCases[n].expected_version));
```

Comment:
It would be nice if somehow the test would print out the difference between expectation and actual values in case of failure.